platforms/core-configuration/model-core/src/main/java/org/gradle/internal/instantiation/generator/Jsr330ConstructorSelector.java

```diff
@@ -16,13 +16,17 @@
 
 package org.gradle.internal.instantiation.generator;
 
+import org.gradle.api.internal.DocumentationRegistry;
 import org.gradle.cache.internal.CrossBuildInMemoryCache;
 import org.gradle.internal.Cast;
+import org.gradle.internal.instantiation.IllegalConstructionException;
 import org.gradle.internal.logging.text.TreeFormatter;
 
 import java.lang.reflect.Modifier;
 
 class Jsr330ConstructorSelector implements ConstructorSelector {
+    static final String CUSTOM_GRADLE_TASK_DOC_LINK = "Rewrite the class with a constructor injection. " + new DocumentationRegistry().getDocumentationRecommendationFor("information on developing custom gradle task types", "custom_tasks");
```

Comment:
❓ I believe the documentation link should be to `custom_gradle_types.html#service_injection` https://docs.gradle.org/current/userguide/custom_gradle_types.html#service_injection
WDYT?

Reply:
thanks for your advice! I have updated the PR